### ISSUE TITLE
Dimension mismatch error when running inference on custom video resolution (1630×970)

### ISSUE THREAD

**Issue by ryanchenggg** [bconstantine/SketchColour#2]
Hi, first of all, thanks for your great work!

I’m trying to run my custom inference on my own video data, but I encounter a tensor size mismatch error during the control pipeline. My input sketch video has resolution 1630×970 with 23 frames at 2 FPS, whereas the repository’s default uses 480×720 resolution with 17 frames at 50 FPS.

**Here's the error log:**
2025-07-15 09:56:17 - [ERROR] - Error during eval: Sizes of tensors must match except in dimension 2. Expected size 6 but got size 5 for tensor number 1 in the list.
…
  File ".../patches/dependencies/diffusers/control.py", line 20, in pre_forward
    control_tensor = torch.cat([original_tensor, input_tensor], dim=dim

**Questions:**
1. Does the model require the input spatial resolution (height × width) to exactly match the default 480×720?
2. Must the number of frames (and frame rate) also be the same as the pretrained setting (17 frames @ 50 FPS)?
3. If custom sizes are supported, what steps (padding, slicing parameters, or kwargs) should I adjust to avoid this mismatch error?

Thank you for any guidance!

**Comment by bconstantine** [comment]
Hi @ryanchenggg ,

Thanks for trying out SketchColour! Happy to help clarify how SketchColour (built on CogVideoX‑I2V‑5B) handles spatial and temporal dimensions:

1. Regarding the spatial size: Our model follows the resolution of CogVideoX‑I2V‑5B which expects fixed 480 × 720 inputs. To avoid tensor‑mismatch, we recommend resizing your sketch frames to 480 × 720 before inference. You can return the outputs back to your original resolution afterward.

2. Frame count: Following CogVideoX requirement as well, the input frames should be 8 N + 1 up to 49 frames (as mentioned in their repository readme). Our pretrained weights is trained under 17 frames. While the model architecture accepts any 8 N + 1 frame length (up to 49), our pretrained weights and evaluations were conducted on 17‑frame inputs. Results with longer or shorter sequences aren’t guaranteed to match the same quality. 

3. Frame rate: FPS only matters for decoding/saving your video. Internally, the model treats frames as a sequence, so you’re free to decode at 2 FPS (or any FPS) and saving at your target rate after generation.

Hope that helps! Feel free to reach out if you have another follow up questions

**Comment by ryanchenggg** [comment]
Thank you for your quick response, which made the entire testing process much smoother!!!

Additionally, I would like to ask about the large motion test data - my results often show motion blur. I'm wondering how you handled this issue, as the demo results appear very stable. Thank you again for your answer.

**Comment by bconstantine** [comment]
Dear @ryanchenggg 

For the test data, we use the Sakuga Dataset’s official test split and run them with the same hyperparameters as in our demo. The demo blur likely comes from how the clips might differ from our test data. Some of the tips that might be useful to make the data distribution similar:
1. Resolution & cropping: We crop frames to maintain aspect ratio instead of naive resizing. Using the same resolution & cropping method can help ensure the model receives inputs at the exact scale and aspect ratio it was trained on, improving consistency and stability.
2. Frame sampling and FPS: Our dataset is a 17-frame video, dominated by 20- 30fps. When encountering long videos, we sample neighboring frames, as uniformly picking 17 frames from a longer clip can shift the motion statistics. 
3. Sketch Similarity and Binarization: We generate binary sketches using InformativeDrawings with a threshold of 250. Matching this binarization ensures your sketches have the same characteristics as our inference inputs.

Hope that helps—let me know if you need any further details!

**Comment by bconstantine** [comment]
As there is no further reply I will close this issue. Feel free to ask follow up questions anytime!
